### PR TITLE
resolve #1009 | disable dynamic reloading of configuration files

### LINKED ISSUE #1009: Disable dynamic reloading of configuration files
Hermes uses archaius library (https://github.com/Netflix/archaius) for configuration management. Current setup allow to dynamically reload properties after changing them in config file. We would like to turn off this feature.

### PR DESCRIPTION
**Pull request by Robert-Duda**
second approach to this topic. pulling archaius initialization from ```ConfigFactory``` out and initializing ```DynamicPropertyFactory``` at the start of ```hermes-consummers``` and ```hermes-fontend```. restore the default way that archaius build configuration with the addition of a configurable option to enable/disable config reload, by default is disabled.

### CODE REVIEW

File: hermes-consumers/src/test/groovy/pl/allegro/tech/hermes/consumers/consumer/offset/OffsetCommitterTest.groovy [hunk 1]
```diff
@@ -353,11 +354,11 @@ class OffsetCommitterTest extends Specification {
         return new SubscriptionPartitionOffset(partitionOffset, offset)
     }
 
-    private assignPartitions(int... partitions) {
+    private assignPartitions(int ... partitions) {
```
Comment (cristaloleg) [inline hermes-consumers/src/test/groovy/pl/allegro/tech/hermes/consumers/consumer/offset/OffsetCommitterTest.groovy]: Looks strange, 'cause `...` are related to the type. Space looks suspicious 🤔

File: hermes-common/src/main/java/pl/allegro/tech/hermes/common/di/factories/ConfigFactoryCreator.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import pl.allegro.tech.hermes.common.config.ConfigFactory;
+
+import static com.netflix.config.ConfigurationManager.APPLICATION_PROPERTIES;
+import static com.netflix.config.ConfigurationManager.DISABLE_DEFAULT_ENV_CONFIG;
+import static com.netflix.config.ConfigurationManager.DISABLE_DEFAULT_SYS_CONFIG;
+import static com.netflix.config.ConfigurationManager.ENV_CONFIG_NAME;
+import static com.netflix.config.ConfigurationManager.SYS_CONFIG_NAME;
+import static com.netflix.config.ConfigurationManager.URL_CONFIG_NAME;
+
+public class ConfigFactoryCreator implements Factory<ConfigFactory> {
```
Comment (dankraw) [inline hermes-common/src/main/java/pl/allegro/tech/hermes/common/di/factories/ConfigFactoryCreator.java]: clever name ;D

File: hermes-consumers/src/test/groovy/pl/allegro/tech/hermes/consumers/consumer/offset/OffsetCommitterTest.groovy (hunk 1 above)
Comment (Robert-Duda) [reply hermes-consumers/src/test/groovy/pl/allegro/tech/hermes/consumers/consumer/offset/OffsetCommitterTest.groovy]: agree, intellij default formatter, I missed it

File: hermes-common/src/main/java/pl/allegro/tech/hermes/common/di/factories/ConfigFactoryCreator.java (hunk 2 above)
Comment (Robert-Duda) [reply hermes-common/src/main/java/pl/allegro/tech/hermes/common/di/factories/ConfigFactoryCreator.java]: its not perfect, but it was not easy to get out from this language puzzle ;-)

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,15 @@
 
 ### ...
 
+## 1.2.5 (27.09.2019)
```
Comment (Robert-Duda) [inline CHANGELOG.md]: @dankraw @druminski 
added an entry to changelog

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,15 @@
...
+
+### Features
+
+#### ([1009](https://github.com/allegro/hermes/issues/1009)) Disable dynamic reloading of configuration files
+Enabling possibility of providing external archaius configuration.
+Pulling archaius initialization from `ConfigFactory` out and initializing `DynamicPropertyFactory` at the start of `hermes-consummers` and `hermes-fontend`. 
+Restore the default way that archaius build configuration with the addition of a configurable option to enable/disable config reload, by default is disabled.
```
Comment (dankraw) [inline CHANGELOG.md]: archaius build -> archaius builds :)

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,15 @@
 
 ### ...
 
+## 1.2.5 (27.09.2019)
+
+### Features
+
+#### ([1009](https://github.com/allegro/hermes/issues/1009)) Disable dynamic reloading of configuration files
+Enabling possibility of providing external archaius configuration.
+Pulling archaius initialization from `ConfigFactory` out and initializing `DynamicPropertyFactory` at the start of `hermes-consummers` and `hermes-fontend`.
```
Comment (dankraw) [inline CHANGELOG.md]: typo: fontend -> frontend

**Pull request merged**